langlib/lang.array/src/main/java/org/ballerinalang/langlib/array/Slice.java

```diff
@@ -77,7 +77,7 @@ public static ArrayValue slice(Strand strand, ArrayValue arr, long startIndex, l
 
         switch (arrType.getTag()) {
             case TypeTags.ARRAY_TAG:
-                slicedArr = new ArrayValueImpl((BArrayType) arrType);
+                slicedArr = new ArrayValueImpl((BArrayType) arrType, sliceSize);
```

Comment:
I think the array slice should be directly supported by ArrayValue. That way we can use java's system.ArrayCopy which is a native JVM implementation and probably be much faster. Shall we create an issue for that.

Reply:
+1 for using `System.arrayCopy()`

Reply:
Created an issue to track this.
https://github.com/ballerina-platform/ballerina-lang/issues/20632